src/Parser/RecursiveTextProcessor.php

```diff
@@ -71,7 +72,7 @@ public function __construct( ?Parser $parser = null ) {
 		$this->parser = $parser;
 
 		if ( $this->parser === null ) {
-			$this->parser = $GLOBALS['wgParser'];
+			$this->parser = ApplicationFactory::getInstance()->create( 'Parser' );
```

Comment:
This seems like it creates a new parser. If the code that uses this value does mutation, then this change could break things.

Reply:
Indeed, but it can also be expected less side-effects (if any) due to unexpected residual state in the parser between different parses; and it be set a parser if desired in some object RecursiveTextProcessor.

I will test in the following days with `@annotation` queries, which are the more extreme SMW+parser subtleties to my knowledge, but I have no idea how to test more.

Also, there are no differences between the [Travis build of the latest merged commit](https://travis-ci.org/github/SemanticMediaWiki/SemanticMediaWiki/builds/771993852) and [this one](https://travis-ci.org/github/SemanticMediaWiki/SemanticMediaWiki/builds/772165201), except for MW 1.35 where there are two errors removed in this build.

Reply:
If it is important to use the global parser, you can use `MediaWikiServices::getParser()` (available since 1.29).

Aside, where on earth is this ApplicationFactory class defined? I spend like 15 minutes looking for it, and could not find it.

Reply:
Sounds frustrating @Nikerabbit. Do you not have class-based search in your editor?

`class_alias( \SMW\Services\ServicesFactory::class, '\SMW\ApplicationFactory' );` in `Aliases.php`. Probably references in SMW itself should be updated if a class is moved...